packages/strapi-design-system/src/DatePicker/DatePickerCalendar.tsx

```diff
@@ -54,98 +74,328 @@ export const DatePickerCalendarContent = ({
+
+  React.useEffect(() => {
     if (selectedDate) {
-      setDate(selectedDate);
+      setLocalDate(convertUTCDateToCalendarDate(selectedDate));
     }
   }, [selectedDate]);
 
-  const handleMonthChange = (month: string) => {
-    const updatedDate = new Date(date);
-    updatedDate.setMonth(months.indexOf(month));
-    setDate(updatedDate);
+  const handleMonthChange = (month: string | number) => {
+    if (typeof month === 'number') {
+      /**
+       * This just to make TS happy, we're not going to get a
+       * number because we only use strings as options
+       */
+      return;
+    }
+
+    const updatedDate = localDate.set({ month: months.indexOf(month) + 1 });
+    setLocalDate(updatedDate);
   };
 
-  const handleYearChange = (year: number) => {
-    const updatedDate = new Date(date);
-    updatedDate.setFullYear(year);
-    setDate(updatedDate);
+  const handleYearChange = (year: string | number) => {
+    if (typeof year === 'number') {
+      /**
+       * This just to make TS happy, we're not going to get a
+       * number because we only use strings as options
+       */
+      return;
+    }
+
+    const updatedDate = localDate.set({ year: parseInt(year, 10) });
+    setLocalDate(updatedDate);
   };
 
-  const handleSelectDay = (date: Date) => () => {
+  const handleSelectDay = (date: DateValue) => () => {
     if (onChange) {
-      onChange(date);
+      onChange(date.toDate(TIME_ZONE));
     }
   };
 
+  const getDatesInWeek = makeGetDatesInWeek(startDate, locale);
+
   return (
-    <Box padding={4}>
-      <Box paddingBottom={4} paddingLeft={2} paddingRight={2}>
-        <Flex>
-          <SimpleMenu label={months[date.getMonth()]}>
-            {months.map((month) => (
-              <MenuItem key={month} onClick={() => handleMonthChange(month)}>
-                {month}
-              </MenuItem>
-            ))}
-          </SimpleMenu>
-          <Box paddingLeft={2}>
-            <SimpleMenu label={date.getFullYear()}>
-              {years.map((year) => (
-                <MenuItem key={year} onClick={() => handleYearChange(year)}>
-                  {year}
-                </MenuItem>
-              ))}
-            </SimpleMenu>
-          </Box>
-        </Flex>
-      </Box>
-      <RawTable colCount={7} rowCount={weeks.length + 1} initialCol={activeCol} initialRow={activeRow} role="grid">
-        <RawThead>
+    <Flex direction="column" alignItems="stretch" padding={4}>
+      <ToolbarFlex justifyContent="flex-start" paddingBottom={4} paddingLeft={2} paddingRight={2} gap={2}>
+        <SingleSelectInput
+          label={monthSelectLabel}
+          size="S"
+          value={months[localDate.month - 1]}
+          onChange={handleMonthChange}
+        >
+          {months.map((month) => (
+            <SingleSelectOption key={month} value={month}>
+              {month}
+            </SingleSelectOption>
+          ))}
+        </SingleSelectInput>
+        <SingleSelectInput
```

Comment:
<img width="382" alt="Screenshot 2023-04-28 at 11 41 47" src="https://user-images.githubusercontent.com/2244375/235113937-db5bed39-cef1-43f4-a06f-94a19ff6030c.png">

For some reason the options of this select have more padding to the right than to the left. Before there was a scrollbar so we had to reserve space, but I think that is no longer needed?

Reply:
> For some reason the options of this select have more padding to the right than to the left. Before there was a scrollbar so we had to reserve space, but I think that is no longer needed?

This is because the `min-width` of the Select dropdown is supposed to be the `min-width` of the select trigger so its technically right, what's weird is we remove the border on these selects, hence why it feels odd.

Reply:
Ah I see. @maevalienard do you think we could render them as standard select components?

Reply:
Instead of simple menus you mean? @gu-stav

Reply:
Simple menu's aren't semantically correct – menus / menu items can't have a value so instead they should be select components – they've been styled as the their simple menu counterparts to match the DS.

Reply:
@maevalienard ping :)